### PR TITLE
[BUGFIX] Fix yarn install and condition on CI pipeline

### PR DESCRIPTION
**Pull request by TrangPham**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: ci/azure-pipelines-docs.yml [hunk 1]
```diff
@@ -43,17 +43,20 @@ stages:
         name: LinkChecker
 
     - job: docs_snippet_checker
-      condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.DocsChanged'], true)
+      condition: or(eq(stageDependencies.scope_check.changes.outputs['CheckDocsChanges.DocsChanged'], true), eq(variables.isDevelop, true), eq(variables.isManual, true))
```
Comment (TrangPham) [inline ci/azure-pipelines-docs.yml:46]: Should not have been changed here: https://github.com/great-expectations/great_expectations/commit/5719dc2a137ff55f6d7f47186e9ddfc7f7c3a3b9#diff-7d7342982960cfff7287cb2365f75648f05069fe1e4e337ec345b6c3409c3d52R46

**Comment by TrangPham** [comment]
It looks like the code got merged without this failing because the check was not automatically ran on the last commit in the PR (https://github.com/great-expectations/great_expectations/pull/8208):
Screenshot 2023-06-29 at 10 36 09 AM
https://github.com/great-expectations/great_expectations/runs/14642699796

Because there was no docs changes within that commit. Wondering if the CI step can look at PR diff rather than the latest commit

File: ci/azure-pipelines-dev.yml [hunk 2]
```diff
@@ -184,14 +184,17 @@ stages:
...
-        name: DocsSnippetChecker
+        - script: |
+            yarn install
+          name: YarnInstall
+        - task: PythonScript@0
+          inputs:
+            scriptSource: 'filePath'
+            scriptPath: 'ci/checks/validate_docs_snippets.py'
+            failOnStderr: false
+          name: DocsSnippetChecker
```
Comment (cdkini) [inline ci/azure-pipelines-dev.yml:197]: Could we make sure this works with a test change before merging? 🙇🏽

Comment (TrangPham) [reply ci/azure-pipelines-dev.yml:197]: Yep! I uncommented so the pipeline would run: https://github.com/great-expectations/great_expectations/pull/8217/checks?check_run_id=14666048161

Screenshot 2023-06-29 at 12 49 04 PM

Passing now

**Review by anthonyburdi — APPROVED** [review]
LGTM! Thanks for the fix.

**Review by cdkini — APPROVED** [review]

**Review by cdkini — APPROVED** [review]

**Pull request merged**